### PR TITLE
cp-170: Add Meditation Player Layout

### PR DESCRIPTION
**Pull request by danilittaa**
- fixed all comment in the issue

- added 'react-native-keep-awake' library;
- removed button toggling during 30-second rewind forward or backward;
- changed some styles;
- changed color when user presses button: 
image

### CODE REVIEW

File: mobile/src/libs/components/button/button.tsx [hunk 1]
```diff
@@ -22,7 +25,30 @@ const Button: React.FC<Properties> = ({
...
+  const [isPressed, setIsPressed] = useState(false);
+
+  let color;
+  if (isPressed) {
+    color = AppColor.BLUE_200;
+  } else {
+    color = isRounded ? AppColor.BLUE_200 : AppColor.GRAY_400;
+  }
+
+  const handleOnPress = (): void => {
```
Comment (dmytro-kucherenko) [inline mobile/src/libs/components/button/button.tsx]:
```suggestion
  const handlePress = (): void => {
```

File: mobile/src/libs/components/player/player.tsx [hunk 2]
```diff
@@ -48,15 +45,25 @@ const Player: React.FC<Properties> = ({ setCurrentTrack }) => {
...
+      KeepAwake.deactivate();
     };
+  }, []);
 
+  useEffect(() => {
+    const addPlaylist = async (): Promise<void> => {
+      void player.setPlaylist(MOCKED_PLAYLIST);
+      setPlaybackState(await player.getState());
+    };
     void addPlaylist();
```
Comment (dmytro-kucherenko) [inline mobile/src/libs/components/player/player.tsx]: Is it necessary to create function to call it once in the next line?

**Review by dmytro-kucherenko — DISMISSED** [review]

File: mobile/src/libs/components/button/button.tsx [hunk 3]
```diff
@@ -22,7 +25,30 @@ const Button: React.FC<Properties> = ({
   type = 'solid',
   isRounded,
   iconName,
+  isVisuallyCentered,
 }) => {
+  const [isPressed, setIsPressed] = useState(false);
+
+  let color;
```
Comment (v0ldemar01) [inline mobile/src/libs/components/button/button.tsx]: Create a separate helper outside for this purpose

File: mobile/src/libs/components/button/button.tsx [hunk 4]
```diff
@@ -22,7 +25,30 @@ const Button: React.FC<Properties> = ({
...
+  }
+
+  const handlePress = (): void => {
+    setIsPressed(true);
+    onPress();
+  };
+
+  useEffect(() => {
+    if (isPressed) {
+      setTimeout(() => {
```
Comment (v0ldemar01) [inline mobile/src/libs/components/button/button.tsx]: What about clearTimeout?

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+const usePressState = ({ onPress, isRounded }: Properties): PressState => {
+  const [isPressed, setIsPressed] = useState(false);
+
+  let color;
+  if (isPressed) {
+    color = AppColor.BLUE_200;
+  } else {
+    color = isRounded ? AppColor.BLUE_200 : AppColor.GRAY_400;
+  }
```
Comment (v0ldemar01) [inline mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts]:
```suggestion
  const color = isPressed || isRounded ? AppColor.BLUE_200 : AppColor.GRAY_400;
```

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import { AppColor } from '#libs/enums/enums';
```
Comment (liza-veis) [inline mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts]: remove this hook, use android ripple instead

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by OlesiaTr — DISMISSED** [review]

File: mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts (hunk 6 above)
Comment (danilittaa) [reply mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts]: I've tried using this before, but the problem is that it specifically changes the background, not the color. However, I just realized that I can use the activeOpacity property, and it looks something like this:

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/108797614/34b707f7-e0bf-4c92-8f88-201b01603323


 What do you think about it?

Comment (liza-veis) [reply mobile/src/libs/hooks/use-press-state/use-press-state.hook.ts]: I like it

**Review by v0ldemar01 — APPROVED** [review]

**Review by iamAlinaaa — APPROVED** [review]
good job)

**Review by liza-veis — APPROVED** [review]

**Pull request merged**